fatiando/gravmag/transform.py

```diff
@@ -503,6 +507,105 @@ def derivz(x, y, data, shape, order=1, method='fft'):
     return deriv[padx: padx + nx, pady: pady + ny].ravel()
 
 
+def power_density_spectra(x, y, data, shape):
+    r"""
+    Calculates the Power Density Spectra of a 2D gridded potential field
+    through the FFT:
+
+    .. math::
+
+        \Phi_{\Delta T}(k_x, k_y) = | F\left{\Delta T \right}(k_x, k_y) |^2
+
+    .. note:: Requires gridded data.
+
+    .. note:: x, y, z and height should be in meters.
+
+    Parameters:
+
+    * x, y : 1D-arrays
+        The x and y coordinates of the grid points
+    * data : 1D-array
+        The potential field at the grid points
+    * shape : tuple = (nx, ny)
+        The shape of the grid
+
+    Returns:
+
+    * kx, ky : 2D-arrays
+        The wavenumbers of each Power Density Spectra point
+    * pds : 2D-array
+        The Power Density Spectra of the data
+    """
+    kx, ky = _fftfreqs(x, y, shape, shape)
+    pds = abs(numpy.fft.fft2(numpy.reshape(data, shape)))**2
+    return kx, ky, pds
+
+
+def radial_average(kx, ky, pds, max_radius=None, ring_width=None):
```

Comment:
OK, I hadn't thought of that. You're right, this has to be specific to something returned by FFT.

I don't think it's worth it trying to make this overly general. It will be a lot of work and there are many subtle ways that bugs can hide in the code. It's better to make it clear that this function only operates on Fourier transforms.

Since this is the case, then it might be better to rename the function to `radial_average_spectrum`  to make it clear that it won't work for general arrays.